cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -791,6 +791,114 @@ Status ConvertListsLike(const PandasOptions& options, const ChunkedArray& data,
   return Status::OK();
 }
 
+inline Status ConvertMap(const PandasOptions& options, const ChunkedArray& data,
+                         PyObject** out_values) {
+  // Get columns of underlying key/item arrays
+  std::vector<std::shared_ptr<Array>> key_arrays;
+  std::vector<std::shared_ptr<Array>> item_arrays;
+  for (int c = 0; c < data.num_chunks(); ++c) {
+    const auto& map_arr = checked_cast<const MapArray&>(*data.chunk(c));
+    key_arrays.emplace_back(map_arr.keys());
+    item_arrays.emplace_back(map_arr.items());
+  }
+
+  const auto& map_type = checked_cast<const MapType&>(*data.type());
+  const auto& key_type = map_type.key_type();
+  const auto& item_type = map_type.item_type();
+
+  // TODO: Is this needed for key/item?
+  /*
+  if (value_type->id() == Type::DICTIONARY) {
```

Comment:
This was taken from the ListConversoin, but not totally sure it applies here to the key/item arrays..

Reply:
Yes, same applies. Fixed and added tests.